panel/tests/dataclasses/test_dataclass.py

```diff
@@ -0,0 +1,505 @@
+    parameterized.age = 3
+    assert parameterized.name == widget.name
+    assert parameterized.age == widget.age
+
+
+def test_widget_viewer_from_class_and_dict_names(model_class, can_observe_field):
+    class ExampleViewer(ModelViewer):
+        _model_class = model_class
+        _model_names = {"name": "xname", "age": "xage"}
+
+    parameterized = ExampleViewer(xage=100)
+
+    widget = parameterized.model
+    assert parameterized.xname == widget.name == "Default Name"
+    assert parameterized.xage == widget.age == 100
+
+    if can_observe_field:
+        # sync: model -> parameterized
+        widget.name = "B"
+        widget.age = 2
+        assert parameterized.xname == widget.name
+        assert parameterized.xage == widget.age
+
+    # widget synced to viewer
+    parameterized.xname = "C"
+    parameterized.xage = 3
+    assert parameterized.xname == widget.name
+    assert parameterized.xage == widget.age
+
+
+def test_widget_viewer_from_instance(model_class, can_observe_field):
+    class ExampleWidgetViewer(ModelViewer):
+        _model_class = model_class
+
+        name = param.String("default", doc="A string parameter")
+        age = param.Integer(3, bounds=(0, 10))
+        not_trait = param.Parameter(1)
+
+    viewer = ExampleWidgetViewer(height=500, sizing_mode="stretch_width")
+    widget = viewer.model
+    assert {"name", "age", "weight", "not_trait", "read_only"} <= set(viewer.param)
+
+    assert viewer.name == widget.name == "Default Name"
+    assert viewer.age == widget.age == 3
+    assert viewer.weight == widget.weight == 0.0
+
+    _assert_is_synced(viewer, widget, can_observe_field)
+
+
+def test_widget_viewer_child_class(model, can_observe_field):
+    class ExampleWidgetViewer(ModelViewer):
+        name = param.String("default", doc="A string parameter")
+        age = param.Integer(3, bounds=(0, 10))
+        not_trait = param.Parameter(-2.0)
+
+    viewer = ExampleWidgetViewer(model=model, height=500, sizing_mode="stretch_width")
+    assert {"name", "age", "weight", "not_trait", "read_only"} <= set(viewer.param)
+
+    assert viewer.name == model.name == "A"
+    assert viewer.age == model.age == 3
+    assert viewer.weight == model.weight == 1.1
+
+    _assert_is_synced(viewer, model, can_observe_field)
+
+
+def test_dont_sync_non_shared_parameter(model):
+    class ExampleWidgetViewer(param.Parameterized):
+        name = param.String("default", doc="A string parameter")
+        age = param.Integer(3, bounds=(0, 10))
+
+    parameterized = ExampleWidgetViewer()
+    sync_with_parameterized(model=model, parameterized=parameterized)
+
+
+def test_dont_sync_non_shared_trait(model):
+    class ExampleWidgetViewer(param.Parameterized):
+        wealth = param.Integer(3, bounds=(0, 10))
+
+    parameterized = ExampleWidgetViewer()
+    sync_with_parameterized(model=model, parameterized=parameterized, names=["wealth"])
+
+def test_model_parameterized_parameters_added_to_instance_not_class(model):
```

Comment:
This test fails. I discovered that `self.param.add_parameter(parameter, param.Parameter())` adds to the class (NOT EXPECTED BY ME) and not the instance.

I need help to understand and fix.